### PR TITLE
React Capstone project - Metrics webapp: Milestone Completion

### PR DESCRIPTION
**Pull request by vanina123**
- Use React documentation.
- Use React components.
- Use React props.
- Use React Router.
- Connect React and Redux.
- Handle events in a React app.
- Write integration tests with a React testing library.
- Use styles in a React app.
- Use React life cycle methods.
- Apply React best practices and language style guides in code.
- Use store, actions and reducers in React.

### CODE REVIEW

File: src/components/homePage.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  const [filteredCharacters, setFilteredCharacters] = useState(characters.results);
+
+  useEffect(() => {
+    dispatch(fetchCharacters());
+  }, [dispatch]);
+
+  useEffect(() => {
+    setFilteredCharacters(characters.results);
+  }, [characters.results]);
+
```
Comment (thecodechaser) [inline src/components/homePage.js:23]: - [ ] You did great on the project but you need to fix the below design issues to follow the given design.
  - Please make sure to display two items(star wars characters) in a row as on the design. Currently, there's one item displayed in a row.

### Your design

image

### Given design

image

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @vanina123👋, 

### You are doing a marvelous job on the React capstone project and this is one of the best projects I have seen so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

image

## To Highlight 🎉
- All tests are passed. ✔️
- No linter errors. :heavy_check_mark:
- Highly organized code. ✔️
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️

- [ ] Please make sure to record a video presenting the features of the project you built and add the video link in the PR description as requested in the project requirements. It is a video with a maximum length of 5 minutes. The content of the video should include:
  - Description of the project.
  - Demo of the project features.
  - You should also highlight some interesting piece of code or something you built that you are very proud of.

- [ ] Check the comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/homePage.js [hunk 2]
```diff
@@ -54,34 +54,36 @@ const HomePage = () => {
...
+                    )}
+                    {character.status === 'unknown' && (
                     <FaCircle className="text-zinc" />
-                  )}
-                  {character.status === 'Dead' && (
+                    )}
+                    {character.status === 'Dead' && (
                     <FaCircle className="text-red" />
-                  )}
-                  {character.status}
```
Comment (fobadara) [inline src/components/homePage.js]: [REQUIRED] Good job dividing the cards into two sections but in order for your design to look more like the template we you need to remove the spaces between and at the side of the cards.

image

**Review by fobadara — CHANGES_REQUESTED** [review]
Hi @vanina123, 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

## Highlights
✅ Implemented reviewer suggestions
✅ Nice design

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by shella12 — APPROVED** [review]
Hi @vanina123, 


## Status : APPROVED 🥳 🥳 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

- Great design implementation ✔️ 
- Great app 👍 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**